spec/paperclip/io_adapters/http_url_proxy_adapter_spec.rb

```diff
@@ -98,4 +98,16 @@
     end
   end
 
+  context "a url with special characters in the filename" do
+    it "returns a file name" do
+      Paperclip::HttpUrlProxyAdapter.any_instance.stubs(:download_content).
+          returns(StringIO.new("x"))
+      url = "https://github.com/thoughtbot/paperclip-öäü字´½♥Ø²È.png"
+      subject = Paperclip.io_adapters.for(url)
+      filename = 'paperclip-%C3%B6%C3%A4%C3%BC%E5%AD%97%C2%B4%C2%BD%E2%99%A5'\
+                 '%C3%98%C2%B2%C3%88.png'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.